task8/src/main/java/PostWork/Service.java

```diff
@@ -0,0 +1,206 @@
+                    new ArrayList<String>(Arrays.asList("#bag")),
+
+            new Post(3,
+                    "Bag number 3, vintage, Italian quality",
+                    "2021-03-17T23:00:00",
+                    "2021-10-17T23:00:00",
+                    "Paseko Nadia",
+                    "https://www.dior.com/en_int/products/couture-M0565OJAP_M885-medium-lady-d-lite-bag-yellow-multicolor-dior-paisley-embroidery",
+                    "10%",
+                    new ArrayList<String>(Arrays.asList("#bag", "#Dior")),
+
+            new Post(4,
+                    "Bag number 4, vintage, Italian quality",
+                    "2021-03-17T23:00:00",
+                    "2021-10-17T23:00:00",
+                    "Paseko Nadia",
+                    "https://www.dior.com/en_int/products/couture-M0565OJAP_M885-medium-lady-d-lite-bag-yellow-multicolor-dior-paisley-embroidery",
+                    "15%",
+                     new ArrayList<String>(Arrays.asList("#bag", "#Dior")),
+
+            new Post(5,
+                    "Bag number 5, vintage, Italian quality",
+                    "2021-03-17T23:00:00",
+                    "2021-10-17T23:00:00",
+                    "Paseko Nadia",
+                    "https://www.dior.com/en_int/products/couture-M0565OJAP_M885-medium-lady-d-lite-bag-yellow-multicolor-dior-paisley-embroidery",
+                    "10%",
+                    new ArrayList<String>(Arrays.asList("#bag", "#Dior")),
+
+            new Post(12,
+                    "Bag number 12, vintage, Italian quality",
+                    "2021-03-17T23:00:00",
+                    "2021-10-17T23:00:00",
+                    "Paseko Nadia",
+                    "https://www.dior.com/en_int/products/couture-M0565OJAP_M885-medium-lady-d-lite-bag-yellow-multicolor-dior-paisley-embroidery",
+                    "15%",
+                    new ArrayList<String>(Arrays.asList("#bag", "#Dior")));
+
+
+    public static Service INSTANCE = null;
+
+    public static Service getInstance() {
+        if (INSTANCE == null) {
+            INSTANCE = new Service();
+        }
+        return INSTANCE;
+    }
+
+
+    public List<Post> getPage(int skip, int top, Map<String, String> filterConfig) {
+        List<Post> filteredPosts = new ArrayList<Post>();
+
+        for (Map.Entry pair : filterConfig.entrySet()) {
+            if (pair.getKey().equals("vendor")) {
+                PList.stream()
+                        .filter(post -> post.getVendor().toLowerCase().contains(pair.getValue().toString().toLowerCase()))
+                        .forEach(filteredPosts::add);
+
+            } else if (pair.getKey().equals("creationDate")) {
+                PList.stream()
+                        .filter(post -> post.getCreatedAt().equals(pair.getValue()))
+                        .forEach(filteredPosts::add);
+            }
+        }
+        if (filterConfig.size() == 0) {
+            filteredPosts = new ArrayList<Post>(PList);
```

Comment:
Just return the original list here. Re-initializing it with another list instance does not have much sence